assembly/src/lib.rs

```diff
@@ -1,45 +1,44 @@
-#![cfg_attr(not(feature = "std"), no_std)]
+#![no_std]
+#![cfg_attr(all(nightly, not(feature = "std")), feature(error_in_core))]
 
-#[cfg(not(feature = "std"))]
 #[macro_use]
 extern crate alloc;
+#[cfg(any(test, feature = "std"))]
+extern crate std;
 
 use vm_core::{
-    code_blocks::CodeBlock,
-    crypto,
+    crypto::hash::RpoDigest,
     errors::KernelError,
     utils::{
-        collections::{btree_map, BTreeMap},
         ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable, SliceReader,
     },
-    CodeBlockTable, Felt, Kernel, Operation, Program, StarkField, ONE, ZERO,
+    Felt, ONE, ZERO,
 };
 
-mod library;
-pub use library::{Library, LibraryNamespace, LibraryPath, MaslLibrary, Module, Version};
-
-mod procedures;
-use procedures::{CallSet, NamedProcedure, Procedure};
-pub use procedures::{ProcedureId, ProcedureName};
+#[cfg(feature = "formatter")]
+pub use vm_core::{prettier, utils::DisplayHex};
 
+mod assembler;
 pub mod ast;
-use ast::{NAMESPACE_LABEL_PARSER, PROCEDURE_LABEL_PARSER};
-
-mod tokens;
-use tokens::{Token, TokenStream};
-
+pub mod diagnostics;
 mod errors;
-pub use errors::{AssemblyError, LabelError, LibraryError, ParsingError, PathError};
-
-mod assembler;
-pub use assembler::{Assembler, AssemblyContext};
-
+pub mod library;
+mod parser;
+pub(crate) mod sema;
```

Comment:
Isn't `pub(crate)` already implied for `sema`? I think we can use just `mod sema;` here.

Reply:
Yeah that didn't use to be the case in earlier versions of Rust, its an old habit of mine. Can definitely be omitted.